tfjs-layers/src/layers/preprocessing/random_width.ts

```diff
@@ -0,0 +1,160 @@
+/**
+ * @license
+ * Copyright 2023 CodeSmith LLC
+ *
+ * Use of this source code is governed by an MIT-style
+ * license that can be found in the LICENSE file or at
+ * https://opensource.org/licenses/MIT.
+ * =============================================================================
+ */
+
+import { image, Rank, serialization, Tensor, tidy } from '@tensorflow/tfjs-core';
+import { getExactlyOneTensor, getExactlyOneShape } from '../../utils/types_utils';
+import {Shape} from '../../keras_format/common';
+import { Kwargs } from '../../types';
+import { ValueError } from '../../errors';
+import { BaseRandomLayerArgs, BaseRandomLayer } from '../../engine/base_random_layer';
+
+export declare interface RandomWidthArgs extends BaseRandomLayerArgs {
+   factor: number | [number, number];
+   interpolation?: InterpolationType; // default = 'bilinear';
+   seed?: number;// default = false;
+   autoVectorize?:boolean;
+}
+
+/**
+ * Preprocessing Layer with randomly varies image during training
+ *
+ * This layer randomly adjusts the width of a batch of images of a
+ * batch of images by a random factor.
+ *
+ * The input should be a 3D (unbatched) or
+ * 4D (batched) tensor in the `"channels_last"` image data format. Input pixel
+ * values can be of any range (e.g. `[0., 1.)` or `[0, 255]`) and of interger
+ * or floating point dtype. By default, the layer will output floats.
+ *
+ * tf methods implemented in tfjs: 'bilinear', 'nearest',
+ * tf methods unimplemented in tfjs: 'bicubic', 'area', 'lanczos3', 'lanczos5',
+ *                                   'gaussian', 'mitchellcubic'
+ *
+ */
+
+const INTERPOLATION_KEYS = ['bilinear', 'nearest'] as const;
+export const INTERPOLATION_METHODS = new Set(INTERPOLATION_KEYS);
+type InterpolationType = typeof INTERPOLATION_KEYS[number];
+
+export class RandomWidth extends BaseRandomLayer {
```

Comment:
[It looks like the keras class extends `BaseImageAugmentationLayer`](https://github.com/keras-team/keras/blob/v2.11.0/keras/layers/preprocessing/image_preprocessing.py#L1898) that (I think) makes some of the indexing calculations generic among layers like this one. Should we do the same here, or is it not applicable in our implementation?

Reply:
I believe the `BaseImageAugmentationLayer` was removed a week and a half ago. [The preprocessing layers extend `BaseRandomLayer`](https://github.com/keras-team/keras/blob/0046e0dcd5b7cac5bd7cc9cb5276f667b80229a8/keras/layers/preprocessing/image_preprocessing.py#L1637) and some of the methodology was moved directly in the preprocessing layers.

Reply:
Thanks for the clarification. In that case, this looks good to me as-is.